data/spells/scripts/support/illusion.lua

```diff
@@ -0,0 +1,33 @@
+local condition = Condition(CONDITION_OUTFIT)
+condition:setTicks(180000)
+
+function onCastSpell(creature, variant)
+	if not creature:isPlayer() then
+		return false
+	end
+
+	local returnValue = RETURNVALUE_NOERROR
+
+	local monsterType = MonsterType(variant:getString())
+	if not monsterType then
+		returnValue = RETURNVALUE_CREATUREDOESNOTEXIST
+	end
+
+	if returnValue == RETURNVALUE_NOERROR then
+		if not getPlayerFlagValue(creature, PlayerFlag_CanIllusionAll) or not monsterType:isIllusionable() then
```

Comment:
I think this could be added as an "else if" branch in line 14. As a consequence, the check for returnValue == RETURNVALUE_NOERROR is not necessary.